contrib/openbsm/bin/bsmconv/bsmconv.c

```diff
@@ -0,0 +1,58 @@
+#include <stdio.h>
+#include <nv.h>
+#include <unistd.h> /* getopt(3) */
+
+#include "linau.h"
+
+#include "pjdlog.h"
+
+int
+main(int argc, char *argv[]) {
+	linau_event *event;
+	linau_record *record;
+	FILE * fp;
+	size_t counter;
+	int debuglevel;
+
+	pjdlog_init(PJDLOG_MODE_STD);
+
+	/* Parse command line options. */
+	debuglevel = 0;
+	for (;;) {
+		int ch;
+
+		ch = getopt(argc, argv, "v");
+		if (ch == -1)
+			break;
+		switch (ch) {
+		case 'v':
+			debuglevel++;
+			break;
+		default:
+			PJDLOG_ABORT("Invalid command line options detected");
+		}
+	}
+
+	pjdlog_debug_set(debuglevel);
+
+	fp = stdin;
+	counter = 0;
+
+	event = linau_event_create();
+	PJDLOG_VERIFY(event != NULL);
+
+	while ((record = linau_record_fetch(fp)) != NULL) {
+		if (!linau_event_empty(event) &&
```

Comment:
The initial id can be invalid so the comparison would return false it an event is empty.